adapters/adapterstest/test_json.go

```diff
@@ -242,19 +249,30 @@ func assertMakeBidsOutput(t *testing.T, filename string, bidderResponse *adapter
 
 // diffHttpRequests compares the actual HTTP request data to the expected one.
 // It assumes that the request bodies are JSON
-func diffHttpRequests(t *testing.T, description string, actual *adapters.RequestData, expected *httpRequest) {
+func diffHttpRequests(description string, actual *adapters.RequestData, expected *httpRequest) error {
+
 	if actual == nil {
-		t.Errorf("Bidders cannot return nil HTTP calls. %s was nil.", description)
-		return
+		return fmt.Errorf("Bidders cannot return nil HTTP calls. %s was nil.", description)
+	}
+
+	if expected.Uri != actual.Uri {
+		return fmt.Errorf(`%s.uri "%s" does not match expected "%s."`, description, actual.Uri, expected.Uri)
 	}
 
-	diffStrings(t, fmt.Sprintf("%s.uri", description), actual.Uri, expected.Uri)
 	if expected.Headers != nil {
-		actualHeader, _ := json.Marshal(actual.Headers)
-		expectedHeader, _ := json.Marshal(expected.Headers)
-		diffJson(t, description, actualHeader, expectedHeader)
+		actualHeader, err := json.Marshal(actual.Headers)
+		if err != nil {
+			return fmt.Errorf(`%s actual.Headers could not be marshalled. Error: %s"`, description, err.Error())
+		}
+		expectedHeader, err := json.Marshal(expected.Headers)
+		if err != nil {
+			return fmt.Errorf(`%s expected.Headers could not be marshalled. Error: %s"`, description, err.Error())
+		}
+		if err := diffJson(description, actualHeader, expectedHeader); err != nil {
+			return err
+		}
 	}
-	diffJson(t, description, actual.Body, expected.Body)
+	return diffJson(description, actual.Body, expected.Body)
```

Comment:
`diffJson` can be fully rerplaced with 
`assert.JSONEq(t, string(expected), string(actualJson))`. Initial simple implementation worked on my local machine right away. It seems to be easier. It was partly replaces in FPD PR, but it looks like some tests still use it.

Reply:
Did it work? I tried `assert.JSONEq(t, string(expected), string(actualJson))` and got many test failures. Where did you substitute it? In which of its calls?

Reply:
Here is my implementation (replace in code there 2 functions)

```

func diffBids(t *testing.T, description string, actual *adapters.TypedBid, expected *expectedBid) {
	if actual == nil {
		t.Errorf("Bidders cannot return nil TypedBids. %s was nil.", description)
		return
	}

	assert.Equal(t, string(expected.Type), string(actual.BidType), fmt.Sprintf(`%s.type "%s" does not match expected "%s."`, description, string(actual.BidType), string(expected.Type)))
	assert.NoError(t, diffOrtbBids(t, fmt.Sprintf("%s.bid", description), actual.Bid, expected.Bid))
}

// diffOrtbBids compares the actual Bid made by the adapter to the expectation from the JSON file.
func diffOrtbBids(t *testing.T, description string, actual *openrtb2.Bid, expected json.RawMessage) error {
	if actual == nil {
		return fmt.Errorf("Bidders cannot return nil Bids. %s was nil.", description)
	}

	actualJson, err := json.Marshal(actual)
	if err != nil {
		return fmt.Errorf("%s failed to marshal actual Bid into JSON. %v", description, err)
	}

	//return diffJson(description, actualJson, expected)
	assert.JSONEq(t, string(expected), string(actualJson), "message123")
	return nil
}


```

Reply:
Oh Gus wait, I tried to replace all of them and it fails. Let me see if I can fix it

Reply:
As discussed offline, this will be done in another PR. Thank you!